zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/FileSystemStorage.java

```diff
@@ -156,7 +158,8 @@ public Void call() throws IOException {
   public synchronized <T> T callHdfsOperation(final HdfsOperation<T> func) throws IOException {
     if (isSecurityEnabled) {
       try {
-        return UserGroupInformation.getCurrentUser().doAs(new PrivilegedExceptionAction<T>() {
+        ugi.reloginFromKeytab();
```

Comment:
Shouldn't this be something like
```
if (UserGroupInformation.isLoginKeytabBased()) {
  UserGroupInformation.getLoginUser().reloginFromKeytab();
} else if (UserGroupInformation.isLoginTicketBased()) {
  UserGroupInformation.getLoginUser().reloginFromTicketCache();
}
```
Ref:https://stackoverflow.com/questions/34616676/should-i-call-ugi-checktgtandreloginfromkeytab-before-every-action-on-hadoop